pilot/pkg/model/push_context.go

```diff
@@ -2315,6 +2315,11 @@ func (ps *PushContext) EnvoyFilters(proxy *Proxy) *MergedEnvoyFilterWrapper {
 func (ps *PushContext) getMatchedEnvoyFilters(proxy *Proxy, namespaces string) []*EnvoyFilterWrapper {
 	matchedEnvoyFilters := make([]*EnvoyFilterWrapper, 0)
 	for _, efw := range ps.envoyFiltersByNamespace[namespaces] {
+		if efw.GetTargetRefs() != nil {
+			// These are meant for a specific target, so we shouldn't treat these as "always match"
+			// In the future, targetRef for EnvoyFilter will likely be implemented -- but currently these would never match
```

Comment:
nit: add a log line so it's observable to the user that their policy was skipped

Reply:
This is the same as a label selector not matching a pod. its not a warnable error that a policy doesn't apply to a specific workload